tkg/client/init.go

```diff
@@ -481,6 +490,60 @@ func (c *TkgClient) MoveObjects(fromKubeconfigPath, toKubeconfigPath, namespace
 	return c.clusterctlClient.Move(moveOptions)
 }
 
+// ApplyClusterBootstrapObjects renders the ClusterBootstrap and its associated objects from templates and applies them to a target management cluster.
+func (c *TkgClient) ApplyClusterBootstrapObjects(fromClusterClient, toClusterClient clusterclient.Client) error {
+	path, err := c.tkgConfigPathsClient.GetTKGProvidersDirectory()
+	if err != nil {
+		return err
+	}
+	clusterBootstrapTemplatesDirPath := filepath.Join(path, "yttcb")
+	configDefaultPath := filepath.Join(path, "config_default.yaml")
+
+	userConfigValuesFile, err := c.getUserConfigVariableValueMapFile()
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		// Remove intermediate config files if err is empty
+		if err == nil {
+			os.Remove(userConfigValuesFile)
+		}
+	}()
+
+	log.V(6).Infof("User ConfigValues File: %v", userConfigValuesFile)
+
+	clusterBootstrapBytes, err := carvelhelpers.ProcessYTTPackage(clusterBootstrapTemplatesDirPath, configDefaultPath, userConfigValuesFile)
+	if err != nil {
+		return err
+	}
+
+	clusterBootstrapString := string(clusterBootstrapBytes)
+	if len(clusterBootstrapString) > 0 {
+
+		// Before applying ClusterBootstrap need to check if TKR is available. Sometimes it takes some time for the
+		// TKr to be available. Hence checking for TKr availability before applying ClusterBootstrap
+		tkr, err := fromClusterClient.GetClusterResolvedTanzuKubernetesRelease()
+		if tkr == nil || err != nil {
+			return err // error occurs or management cluster does not have resolved TKR
+		}
+
+		var toClusterTKR v1alpha3.TanzuKubernetesRelease
+		pollOptions := &clusterclient.PollOptions{Interval: clusterclient.CheckResourceInterval, Timeout: clusterclient.PackageInstallTimeout}
+		log.V(6).Infof("Checking if TKr %s is created on management cluster", tkr.Name)
+		if err := toClusterClient.GetResource(&toClusterTKR, tkr.Name, tkr.Namespace, nil, pollOptions); err != nil {
+			return err
+		}
+
+		log.V(6).Infof("Applying ClusterBootstrap: %v", clusterBootstrapString)
+		if err := toClusterClient.Apply(string(clusterBootstrapBytes)); err != nil {
```

Comment:
I guess `clusterBootstrapString` can be the parameter.

Reply:
fixed